src/pages/MainPage/MainPage.js

```diff
@@ -1,9 +1,240 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
 import './MainPage.scss';
 
 class MainPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      saleCount: 1,
+      insideCount: 1,
+      saleMargin: 0,
+      insideMargin: 0,
+      saleList: [],
+      insideList: [],
+      worldList: [],
+    };
+  }
+
+  handleNextClick = () => {
+    const { saleCount, saleMargin, saleList } = this.state;
+
+    if (saleCount < saleList.length) {
+      this.setState({ saleCount: saleCount + 1 });
+      this.setState({ saleMargin: saleMargin - 421 });
+    }
+  };
+
+  handlePrevClick = () => {
+    const { saleCount, saleMargin, saleList } = this.state;
+
+    if (!saleCount < saleList.length && saleCount > 1) {
+      this.setState({ saleCount: saleCount - 1 });
+      this.setState({ saleMargin: saleMargin + 421 });
```

Comment:
```suggestion
      this.setState({
        saleCount: saleCount - 1,
        saleMargin: saleMargin + 421
      });
```
객체 형태이므로 한번에 업데이트 해주시면 됩니다

Reply:
넵 한 번에 업데이트 형태로 바꿨습니다. 그리고marigin은 state로 관리하지 않고 따로 변수에서 값을 관리하는 걸로 바꿧습니다.